centreon/www/front_src/src/Configuration/ConfigurationBase/Listing/Columns/Status/useStatus.ts

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { ResponseError, useSnackbar } from '@centreon/ui';
+import { useDisable, useEnable } from '../../../api';
+import {
+  labelResourceDisabled,
+  labelResourceEnabled
+} from '../../../translatedLabels';
+
+import { useAtomValue } from 'jotai';
+import { configurationAtom } from '../../../../atoms';
+
+import { capitalize } from '@mui/material';
+
+interface Props {
+  change: (e: React.BaseSyntheticEvent) => void;
+  isMutating: boolean;
+  checked: boolean;
+}
+
+const useStatus = ({ row }): Props => {
+  const { t } = useTranslation();
+
+  const { showSuccessMessage } = useSnackbar();
+
+  const configuration = useAtomValue(configurationAtom);
+  const resourceType = configuration?.resourceType;
+
+  const isActivated = row.isActivated;
+
+  const [checked, setChecked] = useState(isActivated);
+
+  const labelResourceType = capitalize(resourceType as string);
+
+  useEffect(() => {
+    if (isActivated !== checked) {
+      setChecked(isActivated);
+    }
+  }, [isActivated]);
+
+  const { enableMutation, isMutating: isEnableMutating } = useEnable();
+  const { disableMutation, isMutating: isDisableMutating } = useDisable();
+
+  const change = (e: React.BaseSyntheticEvent): void => {
+    const value = e.target.checked;
+    setChecked(value);
+
+    if (checked) {
+      disableMutation({ ids: [row.id] }).then((response) => {
+        const { isError } = response as ResponseError;
+        if (isError) {
+          setChecked(checked);
+
+          return;
+        }
+
+        showSuccessMessage(t(labelResourceDisabled(labelResourceType)));
+      });
+
+      return;
+    }
+
+    enableMutation({ ids: [row.id] }).then((response) => {
+      const { isError } = response as ResponseError;
+      if (isError) {
+        setChecked(checked);
+
+        return;
+      }
+
+      showSuccessMessage(t(labelResourceEnabled(labelResourceType)));
+    });
```

Comment:
Use onSuccess prop of useMutationQuery

Reply:
I will try to enhance the code by extracting the responseAPI function.
But passing onSuccess as a parameter to enableMutation doesn't seem like a good approach. In other use cases, we are updating states like selectedRows and dialogState, which are not the responsibility of the mutation function. These states should be managed separately, as they are unrelated to the core logic of the mutation

Reply:
onSuccess are meant to trigger trigger effects when the mutation succeed

Reply:
I will resolve the conversation if you don't have any other comments on my answer. Thanks